Godeps/_workspace/src/k8s.io/kubernetes/pkg/apiserver/apiserver.go

```diff
@@ -177,7 +177,11 @@ func serviceErrorHandler(requestResolver *APIRequestInfoResolver, apiVersions []
 		}
 	}
 
-	errorJSON(apierrors.NewGenericServerResponse(serviceErr.Code, "", "", "", "", 0, false), codec, response.ResponseWriter)
+	resourceOrKind := requestInfo.Kind
+	if len(resourceOrKind) == 0 {
+		resourceOrKind = requestInfo.Resource
```

Comment:
Another option would be to be able to inject the APIRequestInfoResolver itself into the master config.